fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckTelemetryMiddleware.kt

```diff
@@ -130,6 +134,30 @@ class ReviewQualityCheckTelemetryMiddleware(
             is ReviewQualityCheckAction.OpenPoweredByLink -> {
                 Shopping.surfacePoweredByFakespotLinkClicked.record()
             }
+
+            is ReviewQualityCheckAction.RecommendedProductImpression -> {
+                Shopping.surfaceAdsImpression.record()
+            }
+
+            is ReviewQualityCheckAction.RecommendedProductClick -> {
+                Shopping.surfaceAdsClicked.record()
+            }
+
+            ReviewQualityCheckAction.ToggleProductRecommendation -> {
+                val state = store.state
+                if (state is ReviewQualityCheckState.OptedIn) {
```

Comment:
Idea:
```suggestion
                if (state is ReviewQualityCheckState.OptedIn && state.productRecommendationsPreference != null) {
                 val toggleAction = if (state.productRecommendationsPreference) {
```

We can check that the product recommendation functionality is enabled, then the internal check becomes a bit more cleaner and clearer.